File: src/main/java/com/heybys/oddments/fooddelivery/domain/shop/MenuId.java
package com.heybys.oddments.fooddelivery.domain.shop;

import com.heybys.oddments.base.domain.LongTypeIdentifier;
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType;

public class MenuId extends LongTypeIdentifier {

    public MenuId(Long id) {
        super(id);
    }

    public static class MenuIdJavaType extends LongTypeIdentifierJavaType<MenuId> {
        public MenuIdJavaType() {
            super(MenuId.class);
        }
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/domain/shop/MenuId.kt
package com.heybys.oddments.fooddelivery.domain.shop

import com.heybys.oddments.base.domain.LongTypeIdentifier
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType

class MenuId(id: Long) : LongTypeIdentifier(id) {

    class MenuIdJavaType : LongTypeIdentifierJavaType<MenuId>(MenuId::class.java)
}
